core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +231,323 @@ describe('Editor state service', () => {
+    expect(ecs.getInapplicableSkillMisconceptionIds()).toEqual(['id1', 'id2']);
+  });
+
+  it('should check if current solution is valid', () => {
+    ecs.activeStateName = 'Hola';
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+    solutionValidityService.init(['Hola']);
+    expect(ecs.isCurrentSolutionValid()).toBe(true);
+  });
+
+  it('should delete current solution validity', () => {
+    ecs.activeStateName = 'Hola';
+    solutionValidityService.init(['Hola']);
+    expect(ecs.isCurrentSolutionValid()).toBe(true);
+    ecs.deleteCurrentSolutionValidity();
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+  });
+
+  it('should get event emitter on state editor initialization', () => {
+    let stateEditorInitializedEventEmitter = new EventEmitter<State>();
+    expect(ecs.onStateEditorInitialized).toEqual(
+      stateEditorInitializedEventEmitter);
+  });
+
+  it('should get event emitter on state editor directive' +
+    ' initialization', () => {
+    let stateEditorDirectiveInitializedEventEmitter = new EventEmitter();
+    expect(ecs.onStateEditorDirectiveInitialized).toEqual(
+      stateEditorDirectiveInitializedEventEmitter);
+  });
+
+  it('should get event emitter on interaction editor initialization', () => {
+    let interactionEditorInitializedEventEmitter = new EventEmitter();
+    expect(ecs.onInteractionEditorInitialized).toEqual(
+      interactionEditorInitializedEventEmitter);
+  });
+
+  it('should get event emitter on showing' +
+    ' translation tab busy modal', () => {
+    let showTranslationTabBusyModalEventEmitter = new EventEmitter();
+    expect(ecs.onShowTranslationTabBusyModal).toEqual(
+      showTranslationTabBusyModalEventEmitter);
+  });
+
+  it('should get event emitter on refreshing state translation', () => {
+    let refreshStateTranslationEventEmitter = new EventEmitter();
+    expect(ecs.onRefreshStateTranslation).toEqual(
+      refreshStateTranslationEventEmitter);
+  });
+
+  it('should get event emitter on updating answer choice', () => {
+    let updateAnswerChoicesEventEmitter = new EventEmitter<AnswerChoice[]>();
+    expect(ecs.onUpdateAnswerChoices).toEqual(
+      updateAnswerChoicesEventEmitter);
+  });
+
+  it('should get event emitter on saving outcome destination details', () => {
+    let saveOutcomeDestDetailsEventEmitter = new EventEmitter();
+    expect(ecs.onSaveOutcomeDestDetails).toEqual(
+      saveOutcomeDestDetailsEventEmitter);
+  });
+
+  it('should get event emitter on handling custom arguments update', () => {
+    let handleCustomArgsUpdateEventEmitter = new EventEmitter<AnswerChoice[]>();
+    expect(ecs.onHandleCustomArgsUpdate).toEqual(
+      handleCustomArgsUpdateEventEmitter);
+  });
+
+  it('should get event emitter on object form validity change', () => {
+    let objectFormValidityChangeEventEmitter = new EventEmitter<boolean>();
+    expect(ecs.onObjectFormValidityChange).toEqual(
+      objectFormValidityChangeEventEmitter);
```

Comment:
To test these, can we actually subscribe to these Observables, emit the corresponding event and check that the Observable callback exectutes?

Reply:
Done